File: Android/WatAreMyClasses/app/src/main/java/net/magmastone/NetworkInteraction/Models/WatClass.java
package net.magmastone.NetworkInteraction.Models;

import android.os.Parcelable;

import android.os.Parcel;

/**
 * Created by alex on 8/6/15.
 *
 * This represents a single class returned by the API.
 */

public class WatClass implements Parcelable {
    public  int id;
    public String class_name;
    public String section;
    public  String timestamp;
    public String timeend;
    public String instructor;
    public  String type;
    public  String where;

    protected WatClass(Parcel in) {

        id = in.readInt();
        class_name = in.readString();
        section = in.readString();
        timestamp = in.readString();
        timeend = in.readString();
        instructor = in.readString();
        type = in.readString();
        where = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeString(class_name);
        dest.writeString(section);
        dest.writeString(timestamp);
        dest.writeString(timeend);
        dest.writeString(instructor);
        dest.writeString(type);
        dest.writeString(where);
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<WatClass> CREATOR = new Parcelable.Creator<WatClass>() {
        @Override
        public WatClass createFromParcel(Parcel in) {
            return new WatClass(in);
        }

        @Override
        public WatClass[] newArray(int size) {
            return new WatClass[size];
        }
    };
}

File: iOS/WatAreMyClasses/WatAreMyClasses/WatClass.m
//
//  WatClass.m
//  WatAreMyClasses
//
//  Created by Alex Roth on 2015-08-07.
//  Copyright © 2015 Alex Roth. All rights reserved.
//  This is a basic

#import "WatClass.h"


//This is a basic container that represents a single class.

@implementation WatClass
@synthesize cid,class_name,section,timestamp,timeend,instructor,type,where;

+ (NSDictionary *)JSONKeyPathsByPropertyKey {
    return @{
             @"cid": @"id",
             @"class_name": @"class_name",
             @"timestamp": @"timestamp",
             @"timeend": @"timeend",
             @"instructor": @"instructor",
             @"type": @"type",
             @"where": @"where"
             };
}

@end
